### PR TITLE
Add Normal Mapping into shaders

### PR DESCRIPTION
**Pull request by lowysole**
[Feature] Integrate Normal Mapping in the Editor
[Feature] Integrate Normal Mapping in the shaders (Phong, Specular & Metallic)
[Feature] Add NormalMap strength

Important: You need to delete the local Library/ since now, we are parsing the tangents in the importer from Assimp and we need to reimport the meshes.

Note: Leaks from Parsb solved in https://github.com/PenteractStudios/Tesseract/pull/169

### CODE REVIEW

File: Project/Source/FileSystem/ModelImporter.cpp [hunk 1]
```diff
@@ -126,6 +127,8 @@ static ResourceMesh* ImportMesh(const char* modelFilePath, JsonValue jMeta, cons
 	for (unsigned i = 0; i < assimpMesh->mNumVertices; ++i) {
 		aiVector3D& vertex = assimpMesh->mVertices[i];
 		aiVector3D& normal = assimpMesh->mNormals[i];
+		aiVector3D* tangent = assimpMesh->mTangents;
```
Comment (lowysole) [inline Project/Source/FileSystem/ModelImporter.cpp:130]: Solving one issue here

File: Project/Source/Components/ComponentMeshRenderer.cpp [hunk 2]
```diff
@@ -562,10 +603,10 @@ void ComponentMeshRenderer::Draw(const float4x4& modelMatrix) const {
 		int hasSpecularMap = specular ? 1 : 0;
 
 		// Specular-specific uniform settings
-		if (goBones.size()) {
-			program = App->programs->specularSkinning;
+		if (hasNormalMap) {
+			program = App->programs->specularNormal;
 		} else {
-			program = App->programs->specularNotSkinning;
+			program = App->programs->specularNormal;
```
Comment (JaimeBea) [inline Project/Source/Components/ComponentMeshRenderer.cpp:609]: This should probably be specularNotNormal

**Review by JaimeBea — APPROVED** [review]

File: Project/Source/Components/ComponentMeshRenderer.cpp (hunk 2 above)
Comment (lowysole) [reply Project/Source/Components/ComponentMeshRenderer.cpp:609]: Totally right

**Review by FrancPS — APPROVED** [review]
Fix those issues and its alright! Well done

**Review by Jordi-Gil — APPROVED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Pull request merged**